src/Playwright/Transport/WebSocketTransport.cs

```diff
@@ -0,0 +1,186 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Net.WebSockets;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Playwright.Transport
+{
+    internal class WebSocketTransport : IConnectionTransport, IDisposable
+    {
+        private const int DefaultBufferSize = 1024;  // Byte buffer size
+        private readonly ClientWebSocket _webSocket;
+        private readonly string _wsEndpoint;
+        private readonly ILogger<WebSocketTransport> _logger;
+        private readonly CancellationTokenSource _readerCancellationSource = new();
+        private readonly BrowserTypeConnectOptions _options;
+
+        internal WebSocketTransport(
+            string wsEndpoint = default,
+            BrowserTypeConnectOptions options = default,
+            ILoggerFactory loggerFactory = default)
+        {
+            _logger = loggerFactory?.CreateLogger<WebSocketTransport>();
+            _webSocket = new ClientWebSocket();
+            _wsEndpoint = wsEndpoint;
+            _options = options;
+            SetHeaders();
+        }
+
+        /// <inheritdoc cref="IDisposable.Dispose"/>
+        ~WebSocketTransport() => Dispose(false);
+
+        public event EventHandler<MessageReceivedEventArgs> MessageReceived;
+
+        public event EventHandler<LogReceivedEventArgs> LogReceived;
+
+        public event EventHandler<TransportClosedEventArgs> TransportClosed;
+
+        public bool IsClosed { get; private set; }
+
+        public async Task ConnectAsync()
```

Comment:
See above comment. I wouldn't have any other `public` methods that aren't part of the interface, in this class.

Reply:
Making it a private method & calling in a constructor itself